### PR TITLE
[dxsdk-d3dx/directxsdk] Add message to handle the conflicts between these two ports

### PR DESCRIPTION
**Pull request by Cheney-W**
**Describe the pull request**

- #### What does your PR fix?  
  In our CI test, dxsdk-d3dx failed with following error:
```
  The following files are already installed in D:/installed/x64-windows and are in conflict with dxsdk-d3dx:x64-windows
  Installed by directxsdk:x64-windows
      debug/lib/d3dx10d.lib
      debug/lib/d3dx11d.lib
      debug/lib/d3dx9d.lib
      lib/d3dx10.lib
      lib/d3dx11.lib
      lib/d3dx9.lib
```
For fixing this issue, I add message to handle the conflicts between these two ports .

BTW, directxsdk:x64-windows and directxsdk:x86-windows has been skipped in ci.baseline.txt.

No feature need to be tested.

### CODE REVIEW

**Comment by Neumann-A** [comment]
Stupid question: What exactly is the difference between these two ports? Should there even be two ports?

**Comment by Cheney-W** [comment]
@walbourn 
Do you know more about Neumann-A's question: What exactly is the difference between [dxsdk-d3dx](https://github.com/microsoft/vcpkg/blob/master/ports/dxsdk-d3dx/vcpkg.json) and [directxsdk](https://github.com/microsoft/vcpkg/blob/master/ports/directxsdk/vcpkg.json)? Thanks in advance!

**Comment by walbourn** [comment]
**TL;DR:** The *dxsdk-d3dx* and *directxsdk* ports are *mutually exclusive*, and use of *directxsdk* is strong discouraged.

The legacy DirectX SDK is deprecated, and many of the headers in it conflict with the modern Windows SDK providing much *older* versions of the headers. About the only scenario where the legacy *directxsdk* port is called for is if you are using XAudio 2.7 on Windows XP or the deprecated XACT engine. Using the legacy DirectX SDK adds dependencies on the legacy DirectX End User Redistributable which contains SHA-1 only signed payload DLLs, and XInput 1.3 has a known security bug

See [this blog post](https://walbourn.github.io/where-is-the-directx-sdk-2021-edition/) for the details. This is why the *direcxtsdk* port contains this line:

```
message(WARNING "Build ${PORT} is deprecated, untested in CI, and requires the use of the DirectSetup legacy REDIST solution. See https://aka.ms/dxsdk for more information.")
```

I've done the best I can providing features for the deprecated *directxsdk* port per the instructions in [this blog post](https://walbourn.github.io/the-zombie-directx-sdk/) for the few scenarios where it's still called for, but by in large developers should move away from using it at all.


The recommendations are that new projects should not use either *dxsdk-d3dx* or *directxsdk*, and instead use any of the many open source replacements *directxtk*, *directxtex*, *directxmesh*, *uvatlas*, *directxmath*, etc. per [this blog post](https://walbourn.github.io/living-without-d3dx/) for D3DX. For XAudio2 on Windows 7 or later, the *xaudio2redist* port is the right choice. For XInput, developers should use the original inbox XInput 9.1.0 for Windows 7 or later, or XInput 1.4 for Windows 8 or later because XInput 1.3 has known security bugs, and the feature differences between 9.1.0 are all focused on unused stuff like Xbox 360 headset audio through a controller.


The *dxsdk-d3dx* port exists as a compromise for developers who still make use of the legacy D3DX9/D3DX10/D3DX11 because they have older codebases, are using older tutorials, or simply don't want to rewrite for the new solutions. This port contains only the D3DX headers and has *no header conflicts* with the modern Windows SDK. Furthermore, it includes SHA-256 payload DLLs with a side-by-side license that means the use of this port does not require any use of the legacy DIrectX End User Runtime Redistributable package.

**Review by PhoebeHui — APPROVED** [review]

**Comment by walbourn** [comment]
My recommendation is to move the FATAL_ERROR about the conflict to after the deprecation message, otherwise looks great to me. Thanks.

**Comment by BillyONeal** [comment]
Looks good, thanks for the better error message!
